### DISCUSSION TITLE
How to remove black background (nan values) from loaded raster?

### DISCUSSION THREAD

**Discussion by zxdawn** [opengeos/leafmap#409] [Q&A] [answered]
Hi, I have loaded the raster like this:

```
import rioxarray as rxr
import leafmap.leafmap as leafmap

raster = rxr.open_rasterio(tif_file, mask_and_scale=True)

m = leafmap.Map()

m.add_basemap("Esri.WorldImagery")

m.add_raster(tif_file,
            palette='viridis', layer_name='GHG',
            nodata=raster.rio.nodata, vmin=0,
            opacity=0.7)
```

Is it possible to remove the black background?

![image](https://user-images.githubusercontent.com/30388627/230406953-b6d30935-5d28-418a-87bb-214cadc7f960.png)

**Comment by giswqs** [comment, marked as answer]
The `add_raster` method probably can't understand `raster.rio.nodata`. Try setting the nodata value in the array to another value and use it for the `nodata` parameter for `add_raster`

**Reply by zxdawn** [reply]
Thanks, it works!
